source/shared_patterns/assembly.rb

```diff
@@ -0,0 +1,50 @@
+def assembly_pattern()
+    return PatternRange.new(
+        start_pattern: newPattern(
+            match: /\b(?:__asm__|asm)\b/,
+            tag_as: "storage.type.asm",
+        ).maybe(@spaces).maybe(
+            match: /volatile/,
+            tag_as: "storage.modifier"
+        ).maybe(@spaces).then(
+            match: /\(/,
+            tag_as: "punctuation.section.parens.begin.bracket.round.assembly",
+        ),
+        end_pattern: newPattern(
+            match: /\)/,
+            tag_as: "punctuation.section.parens.end.bracket.round.assembly"
+        ),
+        includes: [
+            PatternRange.new(
+                start_pattern: newPattern(match: /"/, tag_as: "punctuation.definition.string.begin"),
+                end_pattern: newPattern(match: /"/, tag_as: "punctuation.definition.string.end"),
+                tag_as: "string.quoted.double",
+                tag_content_as: "meta.embedded.assembly",
+                includes: [
+                    "source.asm",
+                    "source.x86",
+                    "source.x86_64",
+                    "source.arm",
+                    # this is here because this pattern will not match if all of the includes above are not found.
+                    # it never matches to ensure no overshoot
+                    lookAheadFor(/not/).then(/possible/)
```

Comment:
So what would happen if this was removed?

Reply:
If that pattern is removed, and none of the includeded grammars exist, then the entire string is one token with no scopes.

Reply:
See the spec files in the original commit for what that looks like.

Reply:
Rereading the comment, I realized it is a bit unclear. If the includes consists entirely of includes that cannot be resolved the outer pattern tags will not be applied. If the dummy pattern is removed the entire string gets no special tagging at all.